### PR TITLE
Add notifications on Pokemon spawn

### PR DESCRIPTION
**Pull request by andrewda**
This pull request includes a

- [ ] Bug fix
- [X] New feature
- [ ] Translation

The following changes were made

- Add a notification system
- Add a multi-select to choose Pokémon which, when spawned, will create a notification
- Tested and works in Chrome, Chrome Mobile, and Chromium (including Electron - will fully support Electron ports of PokemonGo-Map)
- Should work on all modern web browsers
- Should work on most mobile browsers (Chrome, Firefox, Android WebView)
- Sound supported in Chrome

This is similar to #1118, but is cleaner and allows custom Pokemon notifications.

### CODE REVIEW

**Comment by JZ6** [comment]
+1 need another colab approval then merge

**Comment by therealshibe** [comment]
Howto set sound?

**Comment by andrewda** [comment]
@therealshibe I don't believe notifications come with a sound by default. I can probably add this later, but it's most likely not top-priority, either.

**Comment by therealshibe** [comment]
Okay I cloned your repo and chrome toast notifications are working for me.

**Comment by ChristianLuxem** [comment]
The localization is missing here, the filter will also show only english Pokemons. I think you should add this..

File: static/map.js [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+
+    if (Notification.permission !== "granted") {
+        Notification.requestPermission();
+    }
+});
+
 var $selectExclude = $("#exclude-pokemon");
+var $selectNotify = $("#notify-pokemon");
 
 $.getJSON("static/locales/pokemon.en.json").done(function(data) {
```
Comment (AHAAAAAAA) [inline static/map.js]: Can we get a:
$.getJSON("static/locales/pokemon."+locale+".json").done(function(data) {

instead, also appropriate changes to app.py and map.html to make this possible!

**Comment by AHAAAAAAA** [comment]
Merged another PR fixing localization, merging this now.

**Comment by MatthiasJ** [comment]
could you tell me, how to use that feature? as command line param or inside gui?
thx

**Comment by HenrySia** [comment]
@andrewda Hi Andrew, i'm pretty sure this worked before, but I did a git pull recently and I couldn't hear the notification sound anymore. Can you perhaps take a quick look? Thank you very much for the good work!

**Pull request merged**